src/test/java/org/sagebionetworks/bridge/spring/controllers/ParticipantDataControllerTest.java

```diff
@@ -0,0 +1,386 @@
+
+        ConsentStatus status = new ConsentStatus.Builder().withName("Name").withGuid(SubpopulationGuid.create("GUID"))
+                .withConsented(true).withRequired(true).withSignedMostRecentConsent(true).build();
+        Map<SubpopulationGuid, ConsentStatus> statuses = Maps.newHashMap();
+        statuses.put(SubpopulationGuid.create(status.getSubpopulationGuid()), status);
+
+        session = new UserSession(participant);
+        session.setAppId(TEST_APP_ID);
+        session.setAuthenticated(true);
+        session.setConsentStatuses(statuses);
+
+        participantData = createParticipantData("a", "b");
+
+        doReturn(app).when(mockAppService).getApp(TEST_APP_ID);
+        doReturn(TEST_USER_ID).when(mockAccount).getId();
+        doReturn(session).when(controller).getSessionIfItExists();
+        doReturn(session).when(controller).getAuthenticatedSession(ADMIN, WORKER);
+        doReturn(session).when(controller).getAuthenticatedAndConsentedSession();
+
+        doReturn(mockRequest).when(controller).request();
+    }
+
+    @Test
+    public void verifyAnnotations() throws Exception {
+        assertCrossOrigin(ParticipantDataController.class);
+        assertGet(ParticipantDataController.class, "getAllDataForSelf");
+        assertGet(ParticipantDataController.class, "getDataByIdentifierForSelf");
+        assertGet(ParticipantDataController.class, "getAllDataForAdminWorker");
+        assertGet(ParticipantDataController.class, "getDataByIdentifierForAdminWorker");
+        assertCreate(ParticipantDataController.class, "saveDataForSelf");
+        assertCreate(ParticipantDataController.class, "saveDataForAdminWorker");
+        assertDelete(ParticipantDataController.class, "deleteDataByIdentifier");
+        assertDelete(ParticipantDataController.class, "deleteDataByIdentifier");
+        assertDelete(ParticipantDataController.class, "deleteDataForAdmin");
+    }
+
+    @Test
+    public void testGetAllDataForSelf() {
+        doReturn(makeResults(OFFSET_KEY, PAGE_SIZE_INT)).when(mockParticipantDataService)
+                .getAllParticipantData(TEST_USER_ID, OFFSET_KEY, PAGE_SIZE_INT);
+
+        ForwardCursorPagedResourceList<String> result = controller.getAllDataForSelf(OFFSET_KEY, PAGE_SIZE_STRING);
+
+        assertEquals(NEXT_PAGE_OFFSET_KEY, result.getNextPageOffsetKey());
+        assertEquals(result.getItems().get(0), IDENTIFIER);
+        assertEquals(result.getItems().get(1), IDENTIFIER);
+    }
+
+    @Test
+    public void testGetAllDataForSelfPageSizeInService() {
+        ArgumentCaptor<Integer> integerCaptor = ArgumentCaptor.forClass(Integer.class);
+        doReturn(makeResults(null, null)).when(mockParticipantDataService)
+                .getAllParticipantData(eq(TEST_USER_ID), eq(null), integerCaptor.capture());
+
+        controller.getAllDataForSelf(null, null);
+
+        assertEquals(API_DEFAULT_PAGE_SIZE, integerCaptor.getValue().intValue());
+    }
```

Comment:
It would be better to use verify. Also, you don't need a captor for the integer. You can just do

```
verify(mockParticipantDataService).getAllParticipantData(TEST_USER_ID, null, API_DEFAULT_PAGE_SIZE)
```